pkg/kubelet/controller.go

```diff
@@ -165,6 +165,16 @@ func (c *Controller) getNodeAddresses(nodes *v1.NodeList) ([]v1.EndpointAddress,
 			errs = append(errs, fmt.Errorf("failed to determine hostname for node (%s): %w", n.Name, err))
 			continue
 		}
+		nodeIsReady := false
+		for _, c := range n.Status.Conditions {
+			if c.Type == v1.NodeReady && c.Status == v1.ConditionTrue {
+				nodeIsReady = true
+				break
+			}
+		}
+		if !nodeIsReady {
+			continue
+		}
```

Comment:
can this be broken out into a helper so this looks more like 

```suggestion
		if !nodeIsReady(n) {
			continue
		}
```